src/Store.tsx

```diff
@@ -123,13 +143,37 @@ const StoreProvider: React.FC<PropsWithChildren> = ({ children }) => {
     init().then(() => console.debug("Storage is initialized"));
   }, []);
 
-  useEffect(() => {
-    if (isLoggedIn) {
-      nmdcServerClient.getCurrentUser().then((user) => setLoggedInUser(user));
-    } else {
-      setLoggedInUser(null);
+  /**
+   * Update the context to reflect the currently logged-in user. This is ultimately determined by
+   * the access token stored in the NmdcServerClient instance. If storage is provided, persist
+   * the user to storage as well.
+   *
+   * @param storage
+   */
+  async function _updateLoggedInUser(storage: Storage | null) {
+    const user = await nmdcServerClient.getCurrentUser();
+    setLoggedInUser(user);
+    if (storage) {
+      await storage.set(StorageKey.LOGGED_IN_USER, user);
```

Comment:
I assume `user` is a "JSON object" as opposed to a string. Based on that assumption, I was surprised to see it being passed to `storage.set` without it being "stringified" first. Maybe `storage.set` takes care of that internally.

I see this statement in the [Capacitor docs](https://capacitorjs.com/docs/v3/apis/storage#working-with-json):
> The Storage API only supports string values. You can, however, use JSON if you `JSON.stringify` the object before calling `set()`, then `JSON.parse` the value returned from `get()`.

Reply:
Oh yeah, this is all very confusing. 

We're not using a Capacitor plugin for storage. The `@capacitor/storage` plugin (quoted above) was renamed to `@capacitor/preferences` a [few releases ago](https://capacitorjs.com/docs/updating/4-0#storage). And that plugin is designed to be a simple key/value store for "lightweight data". That's why we're using [`@ionic/storage`](https://github.com/ionic-team/ionic-storage) instead.

The `@ionic/storage` docs don't make this incredibly clear, but under the hood they use [`localForage`](https://github.com/localForage/localForage) as a storage backend. And `localForage` does [automatically stringify and parse values](https://github.com/localForage/localForage?tab=readme-ov-file#storing-blobs-typedarrays-and-other-js-objects).